src/Rule/Nested.php

```diff
@@ -127,28 +82,8 @@ function (bool $carry, $rule) {
 
     public function getOptions(): array
     {
-        return $this->fetchOptions($this->rules);
-    }
-
-    private function fetchOptions(iterable $rules): array
-    {
-        $result = [];
-        foreach ($rules as $attribute => $rule) {
-            if (is_array($rule)) {
-                $result[$attribute] = $this->fetchOptions($rule);
-            } elseif ($rule instanceof ParametrizedRuleInterface) {
-                $result[$attribute] = $rule->getOptions();
-            } elseif ($rule instanceof RuleInterface) {
-                // Just skip the rule that doesn't support parametrizing
-                continue;
-            } else {
-                throw new InvalidArgumentException(sprintf(
-                    'Rules should be an array of rules that implements %s.',
-                    ParametrizedRuleInterface::class,
-                ));
-            }
-        }
+        $dumper = new RulesDumper();
 
-        return $result;
+        return $dumper->asArray($this->rules, false);
```

Comment:
Why nested rule contain options without rule names? How to handle it without knowing the rules?

Reply:
Because of backward compatibility

Reply:
But, seems, it is senselessly. Return options with rules will use this on frontend and simplify `asArray()` in dumper.

Reply:
I suggest to do each unrelated changes in a different PR.
So if you want to change this logic please create a PR.

Reply:
Now changed `Dumper` in current PR.